pysages/backends/hoomd.py

```diff
@@ -107,11 +107,11 @@ def is_on_gpu(context):
 def take_snapshot(wrapped_context, location=default_location()):
     context = wrapped_context.context
     sysview = wrapped_context.view
-    positions = asarray(positions_types(sysview, location, AccessMode.Read))
-    vel_mass = asarray(velocities_masses(sysview, location, AccessMode.Read))
-    forces = asarray(net_forces(sysview, location, AccessMode.ReadWrite))
-    ids = asarray(rtags(sysview, location, AccessMode.Read))
-    imgs = asarray(images(sysview, location, AccessMode.Read))
+    positions = copy(asarray(positions_types(sysview, location, AccessMode.Read)))
```

Comment:
Is that OK.
What are we copying here.
Just the pointer to the data?
The context manager for the data?
Or a deep copy?

Reply:
This copies the whole array. But this is only called once at the very beginning of the run (this is different than the one called internally in `Sampler`). But I think I can get it to work without calling `copy` here.

Reply:
Solving this differently is going to be trickier than I thought, but this should work fine for now. I'd say we move ahead and find a better way to fix it later.